### PR TITLE
Use builder to create RandomGossipManager

### PR DESCRIPTION
**Pull request by ghost**
All tests passing. No new tests added.

### CODE REVIEW

File: src/main/java/org/apache/gossip/manager/random/RandomGossipManager.java [hunk 1]
```diff
@@ -42,7 +43,14 @@ public static ManagerBuilder newBuilder() {
 
     private ManagerBuilder() {}
 
+    private void checkNotNull(Object arg, String msg) {
```
Comment (doellerbe) [inline src/main/java/org/apache/gossip/manager/random/RandomGossipManager.java]: Why not `Objects.requireNotNull(...)`?

Comment (edwardcapriolo) [inline src/main/java/org/apache/gossip/manager/random/RandomGossipManager.java]: @doellerbe The apache-way is to comment in apache systems, jira, ML. I realize it is a pain when using github. I suggested changing to IllegalArgumentException in the ticket. Commenting here spams the ML so we probably do not want that.

Comment (doellerbe) [inline src/main/java/org/apache/gossip/manager/random/RandomGossipManager.java]: Actually, based on @edwardcapriolo (which I just saw) [comment](https://issues.apache.org/jira/browse/GOSSIP-4?focusedCommentId=15315546&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-15315546), we should keep this, but throw an `IllegalArgumentException`.

**Comment by ghost** [comment]
Closing for now, will reopen with all the changes from review

**Pull request closed without merging**